cl/corpus_importer/tasks.py

```diff
@@ -3053,10 +3062,128 @@ def recap_document_into_opinions(
             cluster.id,
         )
 
+    # Update case name using llm
+    classify_opinion_case_name_task.delay(cluster.pk, recap_document_id)
+
     if not skip_citation_finding:
         find_citations_and_parentheticals_for_opinion_by_pks.delay(
             [opinion.pk]
         )
 
     # Return input task data to preserve the chain in scrape_pacer_free_opinion
     return task_data
+
+
+@app.task(
+    bind=True,
+    max_retries=5,
+    retry_backoff=True,
+    autoretry_for=(
+        RateLimitError,
+        APIConnectionError,
+        ConflictError,
+        InternalServerError,
+        APIError,
+    ),
+    retry_kwargs={"max_retries": 5},
+)
+def classify_opinion_case_name_task(self, pk: int, recap_document_id: int):
+    """Use LLM to extract and verify the correct case name for an opinion ingested from Recap
+
+    This task fetches an OpinionCluster instance, extracts the first 2000 characters of the first
+    related sub_opinion's text, and obtains the docket title from the associated docket
+    object. It sends this data to a shared LLM call function using a cached prompt template
+
+    Upon receiving the LLM json response, it verifies if the document is an opinion and extracts
+    normalized case name information. If valid, it updates the OpinionCluster object with the
+    extracted case name.
+
+    :param self: The Celery task instance
+    :param pk: Primary key of the OpinionCluster object to process
+    :param recap_document_id: RECAPDocument id
+    """
+
+    system_prompt_path = (
+        "cl/corpus_importer/prompts/case_name_extract_system.txt"
+    )
+    user_prompt_path = "cl/corpus_importer/prompts/case_name_extract_user.txt"
```

Comment:
We should have a better way to import prompts. Notionally, it could be something like:

```
sys_prompt, user_prompt = get_prompts("case_name_extractor")
```

Or maybe:
```
sys_prompt = get_prompt("case_name_extractor", type="system")
user_prompt = get_prompt("case_name_extractor", type="user")
```

Not sure which is right or something else, but both are better than big long paths in our code. 

And, just stylistically, we should always use the `Path` module for this kind of thing too. If you ever see something like `some/path/to/a/file` it's a bug.